src/framework/handlers/esmscript.js

```diff
@@ -0,0 +1,97 @@
+export class ScriptCache {
+    static _scripts = new Map();
+
+    static register(path, script) {
+        const url = URL.canParse(path) ? new URL(path) : new URL(path, 'file://');
+        ScriptCache._scripts.set(url.pathname, script);
+    }
+
+    static get(path) {
+        const url = URL.canParse(path) ? new URL(path) : new URL(path, 'file://');
+        return ScriptCache._scripts.get(url.pathname);
+    }
+
+    static clear() {
+        ScriptCache._scripts.clear();
+    }
+}
+
+/**
+ * This custom import statement for ES Modules conditionally
+ * checks and resolves modules in bundled or non bundled modules
+ *
+ * **The api is likely to change, use at your own discretion**
+ *
+ * @param {import('../app-base').AppBase} app - The application scope to load from.
+ * @param {string} moduleSpecifier - A unique path or specifier used to import the module
+ * @returns {Promise} Returns a promise which fulfills to a module namespace object: an object containing all exports from moduleName.
+ *
+ * @todo add support for bundle contexts
+ * @ignore
+ */
+export const DynamicImport = (app, moduleSpecifier) => {
+    // eslint-disable-next-line multiline-comment-style
+    /* #if _ASSET_BASE_URL
```

Comment:
is this dead code or ...?

Reply:
No, this is a compile time flag necessary when running the editor with  `use_local_engine`. Without this the import is relative to the  custom engine which would be localhost. Here we would need to set `ASSET_BASE_URL=https://playcanvas.com` etc

Reply:
Feels like this should be a runtime config instead of compile time. Otherwise we'd have to use a version of the engine specifically compiled for each domain?

Reply:
It's only ever used during development when running `npm run develop` and this gets stripped during production build. I was reluctant to include any runtime code unless it was needed in prod

Reply:
We actually use `use_local_engine` quite regularly on dev and prod to test things, it's how we test engine releases. Currently we can use any engine for the editor or launch page.

Reply:
Makes sense. In that case this will need to be runtime. We currently have the `AssetRegistry.prefix`, which seems like the most sensible place for this, but it's currently a relative path 🤔